src/BuildScriptGenerator/Hugo/StaticSiteGeneratorHelper.cs

```diff
@@ -112,11 +113,15 @@ public static bool IsHugoApp(ISourceRepo sourceRepo, IEnvironment environment)
             foreach (var path in yamlFilePaths)
             {
                 var yamlNode = ParserHelper.ParseYamlFile(sourceRepo, path);
-                if (yamlNode.Children.Keys
-                    .Select(k => k.ToString())
-                    .Any(k => HugoConfigurationVariables.Contains(k, StringComparer.OrdinalIgnoreCase)))
+                var yamlMappingNode = yamlNode as YamlMappingNode;
+                if (yamlMappingNode != null)
```

Comment:
If it is not a yamlMappingNode but a yamlSequenceNode, we probably still want to look into the sequence and get document for each item?